crates/pipeline_manager/src/db.rs

```diff
@@ -789,77 +888,125 @@ impl ProjectDB {
     pub(crate) async fn get_config(&self, config_id: ConfigId) -> AnyResult<ConfigDescr> {
-        let descr: Result<ConfigDescr, DBError> = sqlx::query(
-            "SELECT project_id, version, name, config FROM project_config WHERE id = $1",
+        let (mut descr, pipeline_id): (ConfigDescr, Option<PipelineId>) = sqlx::query(
+            "SELECT id, version, name, description, config, pipeline_id, project_id FROM project_config WHERE id = ?",
         )
         .bind(config_id.0)
         .fetch_one(&self.conn)
         .await
         .map(|row| {
-            Ok(ConfigDescr {
+            let pipeline_id: Option<PipelineId> = row.get::<Option<i64>, _>(5).map(PipelineId);
+            let project_id = row.get::<Option<i64>, _>(6).map(ProjectId);
+
+            (ConfigDescr {
                 config_id,
-                project_id: ProjectId(row.get(0)),
+                project_id,
                 version: Version(row.get(1)),
                 name: row.get(2),
-                config: row.get(3),
-            })
+                description: row.get(3),
+                config: row.get(4),
+                pipeline: None,
+                attached_connectors: Vec::new(),
+            }, pipeline_id)
         })
         .map_err(|_| anyhow!(DBError::UnknownConfig(config_id)))?;
-        Ok(descr?)
+
+        descr.attached_connectors = self.get_attached_connectors(config_id).await?;
+        if let Some(pipeline_id) = pipeline_id {
+            descr.pipeline = Some(self.get_pipeline(pipeline_id).await?);
+        }
+
+        Ok(descr)
     }
 
     /// Create a new project config.
     pub(crate) async fn new_config(
         &self,
-        project_id: ProjectId,
+        project_id: Option<ProjectId>,
         config_name: &str,
+        config_description: &str,
         config: &str,
+        connectors: &Option<Vec<AttachedConnector>>,
     ) -> AnyResult<(ConfigId, Version)> {
-        debug!("new_config {project_id} {config_name} {config}");
-        // Check that the project exists, so we return correct error status
-        // instead of Internal Server Error due to the next query failing.
-        let _descr = self.get_project(project_id).await?;
-
         let row = sqlx::query(
-            "INSERT INTO project_config (project_id, version, name, config) VALUES($1, 1, $2, $3) RETURNING id")
-            .bind(project_id.0)
+            "INSERT INTO project_config (project_id, version, name, description, config) VALUES(?, 1, ?, ?, ?) RETURNING id")
+            .bind(project_id.map(|id| id.0))
             .bind(config_name)
+            .bind(config_description)
             .bind(config)
             .fetch_one(&self.conn)
             .await?;
-        Ok((ConfigId(row.get(0)), Version(1)))
+        let config_id = ConfigId(row.get(0));
+
+        if let Some(connectors) = connectors {
+            // Add the connectors.
+            // TODO: This should be done in a transaction with the query above.
```

Comment:
How does it work, is this whole function executed in a transaction? @lalithsuresh @ryzhyk ?